site/docs/main/httpproxy.md

```diff
@@ -403,6 +403,74 @@ spec:
 In this example, the permission for Contour to reference the Secret `example-com-wildcard` in the `admin` namespace has been delegated to HTTPProxy objects in the `example-com` namespace.
 Also, the permission for Contour to reference the Secret `another-com-wildcard` from all namespaces has been delegated to all HTTPProxy objects in the cluster.
 
+#### CORS policy
+
+A CORS (Cross-origin resource sharing) policy can be set for a HTTPProxy in order to allow cross-domain requests for trusted sources. If a policy is set, it will be applied to all the routes of the virtual host.
+
+Contour allows configuring the headers involved in cross-domain requests. In this example, cross-domain requests will be allowed for any domain (note the `*` value).
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+spec:
+  virtualhost:
+    fqdn: www.example.com
+  corsPolicy:
+      allowCredentials: true
+      allowOrigin: 
+        - "*" # allows any origin
+      allowMethods:
+        - GET
+        - POST
+        - OPTIONS
+      allowHeaders: 
+        - authorization
+        - cache-control
+      exposeHeaders: 
+        - Content-Length
+        - Content-Range
+      maxAge: "10m" # preflight requests can be cached for 10 minutes. 
+  routes:
+    - conditions:
+      - prefix: /
+      services:
+        - name: s1
+          port: 80
+```
+
+In the following example, cross-domain requests are restricted to https://client.example.com only.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+spec:
+  virtualhost:
+    fqdn: www.example.com
+  corsPolicy:
```

Comment:
This should be indented one more to be inside the `virtualhost` struct.

Reply:
good catch!